test/e2e/user_certs_test.go

```diff
@@ -223,6 +236,22 @@ func TestNamedCertificates(t *testing.T) {
 	}
 }
 
+func apiserverPodRevision(kubeClient *clientcorev1.CoreV1Client) (int, error) {
+	var revision int
+	err := retry.RetryOnConnectionErrors(context.Background(), func(ctx context.Context) (bool, error) {
```

Comment:
I think you want https://github.com/kubernetes/client-go/blob/master/tools/watch/until.go#L127

Reply:
I took a stab at it. PTAL.